### DISCUSSION TITLE
Code hints for `*args, **kwargs` passed to function

### DISCUSSION THREAD

**Discussion by ItsCubeTime** [microsoft/pylance-release#5677] [Enhancement]
In the following snippet:

![image](https://github.com/microsoft/pylance-release/assets/20190653/de08b0dc-c3cb-4067-bc76-ef931b2e1c9a)
```py
class DummyBaseBase:
    def __init__(self, c, *args, **kwargs) -> None:
        print(f"args: {args}")
        print(c)
class DummyBase(DummyBaseBase):
    def __init__(self, b, *args, **kwargs) -> None:
        print(b)
        super().__init__(*args,**kwargs)
class Dummy(DummyBase):
    def __init__(self, a, *args, **kwargs) -> None:
        print(a)
        super().__init__(*args,**kwargs)


dummy=Dummy(a=1,b=2,c=3)
```
When instantiating `Dummy`, autocompletion currently only suggests the immediate parameters (in this case `a, *args, **kwargs`). Would be handy if you could also see the parameter names of any functions called inside `Dummy.__init__` (and any functions called inside those functions) that makes use of `*args, **kwargs`.

So, **instead of hinting** `a,*args,**kwargs`, **it would hint** `a,b,c,*args,**kwargs`

**Comment by debonte** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
